pkg/virt-controller/watch/application_test.go

```diff
@@ -45,6 +64,108 @@ func newValidGetRequest() *http.Request {
 var _ = Describe("Application", func() {
 	var app VirtControllerApp = VirtControllerApp{}
 
+	It("Reports leader prometheus metric when onStartedLeading is called ", func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient := kubecli.NewMockKubevirtClient(ctrl)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		vmInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachine{})
+		vmiInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstance{})
+		vmSnapshotInformer, _ := testutils.NewFakeInformerFor(&snapshotv1.VirtualMachineSnapshot{})
+		vmSnapshotContentInformer, _ := testutils.NewFakeInformerFor(&snapshotv1.VirtualMachineSnapshotContent{})
+		migrationInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstanceMigration{})
+		nodeInformer, _ := testutils.NewFakeInformerFor(&kubev1.Node{})
+		recorder := record.NewFakeRecorder(100)
+		config, _, _, _ := testutils.NewFakeClusterConfig(&kubev1.ConfigMap{})
+		pdbInformer, _ := testutils.NewFakeInformerFor(&v1beta1.PodDisruptionBudget{})
+		podInformer, _ := testutils.NewFakeInformerFor(&k8sv1.Pod{})
+		pvcInformer, _ := testutils.NewFakeInformerFor(&k8sv1.PersistentVolumeClaim{})
+		dataVolumeInformer, _ := testutils.NewFakeInformerFor(&cdiv1.DataVolume{})
+		rsInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstanceReplicaSet{})
+		storageClassInformer, _ := testutils.NewFakeInformerFor(&storagev1.StorageClass{})
+		crdInformer, _ := testutils.NewFakeInformerFor(&extv1beta1.CustomResourceDefinition{})
+		vmRestoreInformer, _ := testutils.NewFakeInformerFor(&snapshotv1.VirtualMachineRestore{})
+
+		var qemuGid int64 = 107
```

Comment:
This isn't defined anywhere in production code?

Reply:
Do you mean the `qemuGid`? We have a default in `application.go` called `defaultLauncherSubGid`. The test doesn't rely on this.